### PR TITLE
Fix crud tests

### PR DESCRIPTION
**Pull request by azarek**

### CODE REVIEW

**Comment by Archbold** [comment]
As with product code changes any automation code changes should go from the branch, to preview then to master. :) Don't worry about this PR but do for future ones. Thanks

**Comment by Archbold** [comment]
LGTM I.e. running the CRUD tests from the branch has no fails, the tests are still testing the test cases and there is no apparent of regression of other test cases.

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/EditMode.java [hunk 1]
```diff
@@ -48,12 +50,29 @@ private void submit() {
...
+    wait.forElementNotPresent(submitButtonBy);
+    PageObjectLogging.logInfo("Page submitted");
+
+    return new ArticlePageObject();
+  }
+
+  public ArticlePageObject submitArticleAsAnon() {
+    submit();
+
+    wait.forElementVisible(notificationForAnon);
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/EditMode.java]: Add javadoc with explanation why we need to publish two times

File: src/test/java/com/wikia/webdriver/testcases/articlecrudtests/ArticleCRUDAnonTests.java [hunk 2]
```diff
@@ -57,37 +55,34 @@ public void ArticleCRUDAnon_003_addDropdown() {
...
   public void ArticleCRUDAnon_006_editDropdown() {
     String articleContent = PageContent.ARTICLE_TEXT;
     ArticlePageObject article = new ArticlePageObject().open("AnonEditDropdown");
     VisualEditModePageObject visualEditMode = article.editArticleInCKUsingDropdown();
     visualEditMode.addContent(articleContent);
-    visualEditMode.clickPublishButton();
+    visualEditMode.submitArticleAsAnon();
     article.verifyContent(articleContent);
   }
 }
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/testcases/articlecrudtests/ArticleCRUDAnonTests.java]: Add a test for notification not appearing for the second time an anon edits a page

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/EditMode.java [hunk 3]
```diff
@@ -48,12 +50,29 @@ private void submit() {
...
 
   public ArticlePageObject submitArticle() {
     submit();
+    wait.forElementNotPresent(submitButtonBy);
+    PageObjectLogging.logInfo("Page submitted");
+
+    return new ArticlePageObject();
+  }
+
+  public ArticlePageObject submitArticleAsAnon() {
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/EditMode.java]: Submitting as anon works this way only the first time we edit. Maybe go with "submitExpectingNotification" and then a regular submit. This way we won't hide test flow in test body ;)

File: src/test/java/com/wikia/webdriver/testcases/articlecrudtests/ArticleCRUDAnonTests.java [hunk 4]
```diff
@@ -57,37 +55,48 @@ public void ArticleCRUDAnon_003_addDropdown() {
...
     ArticlePageObject article = new ArticlePageObject().open("AnonEditDropdown");
     VisualEditModePageObject visualEditMode = article.editArticleInCKUsingDropdown();
     visualEditMode.addContent(articleContent);
-    visualEditMode.clickPublishButton();
+    visualEditMode.submitExpectingNotification().submitArticle();
     article.verifyContent(articleContent);
   }
+
+  @Test(groups = {"ArticleCRUDAnon_007"})
+  public void ArticleCRUDAnon_007_editArticleSecondTime() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/articlecrudtests/ArticleCRUDAnonTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/testcases/articlecrudtests/ArticleCRUDAnonTests.java (hunk 4 above)
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/testcases/articlecrudtests/ArticleCRUDAnonTests.java]: please rename to match JAVA guidelines

File: src/test/java/com/wikia/webdriver/testcases/articlecrudtests/ArticleCRUDAnonTests.java [hunk 5]
```diff
@@ -57,37 +55,48 @@ public void ArticleCRUDAnon_003_addDropdown() {
...
     visualEditMode.addContent(articleContent);
-    visualEditMode.submitArticle();
+    visualEditMode.submitExpectingNotification().submitArticle();
     article.verifyContent(articleContent);
   }
 
   @Test(groups = {"ArticleCRUDAnon_006"})
-  @RelatedIssue(issueID = "MAIN-6278", comment = "test manually")
-  public void ArticleCRUDAnon_006_editDropdown() {
+  public void articleCRUDAnon_editDropdown() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/articlecrudtests/ArticleCRUDAnonTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Pull request merged**